File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[,] score) {
        int[] answer = new int[score.GetLength(0)]; 
        
        List<double> list = new List<double>();
        
        for(int i = 0; i < score.GetLength(0); i++)
        {
            double average = (score[i, 0] + score[i, 1]) / 2.0;
            list.Add(average);
        }
        
        for(int i = 0; i < score.GetLength(0); i++)
        {
            int rank = 1;
            for(int j = 0; j < score.GetLength(0); j++)
            {
                if(list[i] < list[j]) rank++;
            }
            answer[i] = rank;
        }
        return answer;
    }
}

File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<vector<int>> score) {
    vector<int> answer;
    vector<int> average;
    for(int i = 0; i < score.size(); i++)
    {
        int avg = (score[i][0] + score[i][1]);
        average.push_back(avg);
    }
    
    for(int i = 0; i < average.size(); i++)
    {
        int rank = 1;
        for(int j = 0; j < average.size(); j++)
        {
            if(average[j] > average[i])
                rank++;
        }
        answer.push_back(rank);
    }
    return answer;
}